iam.tf

```diff
@@ -0,0 +1,119 @@
+# -------------------------------------------------------------------------------------------------
+# IAM Policy Document
+# -------------------------------------------------------------------------------------------------
+data "aws_iam_policy_document" "assume_role_service" {
+  count = "${var.iam_role_enabled ? 1 : 0}"
+
+  statement {
+    effect  = "Allow"
+    actions = ["sts:AssumeRole"]
+
+    principals {
+      type        = "Service"
+      identifiers = ["ec2.amazonaws.com"]
+    }
+  }
+}
+
+data "aws_iam_policy_document" "assume_role_principals" {
+  count       = "${var.iam_role_enabled ? signum( ( var.iam_user_enabled ? 1 : 0 ) + length(var.iam_role_principals_arns)) : 0}"
+  source_json = "${element(concat(data.aws_iam_policy_document.assume_role_service.*.json, list("")), 0)}"
+
+  statement {
+    effect  = "Allow"
+    actions = ["sts:AssumeRole"]
+
+    principals {
+      type        = "AWS"
+      identifiers = ["${concat(var.iam_role_principals_arns, aws_iam_user.this.*.arn)}"]
+    }
+  }
+}
+
+# -------------------------------------------------------------------------------------------------
+# IAM Role
+# -------------------------------------------------------------------------------------------------
+resource "aws_iam_role" "this" {
+  count = "${var.iam_role_enabled ? 1 : 0}"
+  name  = "${var.env}-microservice-${var.name}"
+
+  assume_role_policy = "${length(var.iam_role_principals_arns) + ( var.iam_user_enabled ? 1 : 0 ) > 0 ?
+		element(concat(data.aws_iam_policy_document.assume_role_principals.*.json, list("")), 0) :
+		data.aws_iam_policy_document.assume_role_service.json}"
+
+  description = "IAM Role for ${var.env}-microservice-${var.name}"
+
+  tags = "${local.tags}"
+}
+
+# -------------------------------------------------------------------------------------------------
+# IAM User
+# -------------------------------------------------------------------------------------------------
+resource "aws_iam_user" "this" {
+  count         = "${var.iam_user_enabled ? 1 : 0}"
+  name          = "${var.env}-microservice-${var.name}"
+  path          = "${var.iam_user_path}"
+  force_destroy = "true"
+  tags          = "${local.tags}"
+}
+
+resource "aws_iam_access_key" "this" {
+  count = "${var.iam_user_enabled ? 1 : 0}"
+  user  = "${aws_iam_user.this.name}"
+}
+
+# -------------------------------------------------------------------------------------------------
+# IAM DynamoDB resources
+# -------------------------------------------------------------------------------------------------
+data "aws_iam_policy_document" "dynamodb_full_access" {
+  count = "${var.dynamodb_enabled && var.iam_role_enabled ? 1 : 0}"
+
+  statement {
+    sid    = "ListDynamoDB"
+    effect = "Allow"
+
+    resources = [
+      "*",
+    ]
+
+    actions = [
+      "dynamodb:List*",
```

Comment:
This permission requires explanation. Why does access to one DynamoDB instance imply those kinds of permissions to all other available DynamoDB instances

Reply:
https://docs.aws.amazon.com/IAM/latest/UserGuide/reference_policies_examples_dynamodb_specific-table.html Can we leave it as is for now ?